camel-idea-plugin/src/main/java/com/github/cameltooling/idea/service/extension/camel/JavaCamelIdeaUtils.java

```diff
@@ -53,6 +56,35 @@
 
     private static final String JAVA_LANG_STRING = "java.lang.String";
 
+    @Override
+    public boolean isCamelFile(PsiFile file) {
+        final List<String> builders = Arrays.asList(
+                new String[] {
+                        "RouteBuilder",
+                        "RoutesBuilder",
+                        "RouteConfigurationBuilder",
+                        "RouteConfigurationsBuilder",
+                        "AdviceWithRouteBuilder",
+                        "EndpointRouteBuilder",
+                        "EndpointRouteConfigurationBuilder",
+                        "SpringRouteBuilder"
+                });
+        if (file != null && file.getFileType().equals(JavaFileType.INSTANCE)) {//This is Java file
```

Comment:
better practice to invert equals to avoid potential NPE in case getFileType returns null